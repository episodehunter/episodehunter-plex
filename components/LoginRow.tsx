import * as React from 'react';

export enum StatusType {
  Ok,
  Warning,
  Error
};

interface LoginRowArgs {
  status: StatusType;
  buttonText: string;
  onButtonClick: Function;
}

function StatusIcon(props: {status: StatusType}) {
  const style = {
    borderRadius: '10px',
    width: '20px',
    height: '20px',
    alignSelf: 'center',
    marginRight: '10px',
    backgroundColor: ['green', '#d2d200', 'red'][props.status]
  };
  return (
    <div style={style} />
  );
}

function LogInButton(props: {buttonText: string, onClick: (e) => void}) {
  return (
    <button onClick={props.onClick}>{props.buttonText}</button>
  );
}

export function LoginRow(props: {status: StatusType, buttonText: string, onButtonClick: (e) => void, errorMsg?: string}) {
  const style = {
    display: 'flex',
    paddingTop: '20px',
    flexWrap: 'wrap'
  } as any;
  return (
    <div style={style}>
      <StatusIcon status={props.status} />
      <LogInButton buttonText={props.buttonText} onClick={props.onButtonClick}/>
      {props.errorMsg}
    </div>
  );
}
