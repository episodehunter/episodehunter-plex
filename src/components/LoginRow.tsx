import * as React from 'react';
import { StatusIcon } from './StatusIcon';
import { LogInButton } from './LoginButton';
import { StatusType } from '../types';

const style = {
  display: 'flex',
  paddingTop: '20px',
  flexWrap: 'wrap'
} as any;

export function LoginRow(props: {status: StatusType, buttonText: string, onButtonClick: (e) => void, errorMsg?: string}) {
  return (
    <div style={style}>
      <StatusIcon status={props.status} />
      <LogInButton buttonText={props.buttonText} onClick={props.onButtonClick}/>
      {props.errorMsg}
    </div>
  );
}
