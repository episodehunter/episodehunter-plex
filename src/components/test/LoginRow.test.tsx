import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { LoginRow } from '../LoginRow';

test('that LoginRow renders', () => {
  const tree = renderer.create(
    <LoginRow status={0} buttonText="Button text!" onButtonClick={() => {}} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
