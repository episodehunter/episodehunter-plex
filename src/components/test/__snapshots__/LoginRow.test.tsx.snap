// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`that LoginRow renders 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexWrap": "wrap",
      "paddingTop": "20px",
    }
  }
>
  <div
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "green",
        "borderRadius": "10px",
        "height": "20px",
        "marginRight": "10px",
        "width": "20px",
      }
    }
  />
  <button
    onClick={[Function]}
  >
    Button text!
  </button>
</div>
`;
